share/models/sources.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.db import models
+
+from share.models.ingest import SourceConfig
+from share.models.ingest import NaturalKeyManager
+
+logger = logging.getLogger(__name__)
+__all__ = ('SourceStat', 'SourceStatOAI',)
+
+
+class SourceStat(models.Model):
+    config = models.ForeignKey(SourceConfig, on_delete=models.CASCADE)
+    is_deleted = models.BooleanField(default=False)
+    date_created = models.DateTimeField(auto_now_add=True)
+    response_status_code = models.SmallIntegerField(blank=True, null=True)
+    response_elapsed_time = models.FloatField(blank=True, null=True)
+    response_exception = models.TextField(blank=True, null=True)
+    earliest_datestamp_config = models.DateField(blank=True, null=True)
+    base_url_config = models.TextField()
+    admin_note = models.TextField(blank=True)
+
+    objects = NaturalKeyManager('config.label')
+
+    def natural_key(self):
+        return (self.config.label,)
+
+    def __repr__(self):
+        return '<{}({}, {})>'.format(self.__class__.__name__, self.pk, self.config.label)
+
+    def __str__(self):
+        return '{}: {}'.format(self.config.source.long_title, self.config.label)
+
+
+class SourceStatOAI(SourceStat):
```

Comment:
:( Yeah...it doesn't make any sense to do it that way. Didn't really think that through...